src/document/docUtil.ts

```diff
@@ -27,32 +28,34 @@ export function getPageNumber(element: Element | null): number | undefined {
     return pageNumber ? parseInt(pageNumber, 10) : undefined;
 }
 
-export function getRange(): Range | null {
+export function getSelection(): Selection | null {
     const selection = window.getSelection();
 
-    if (!selection || selection.isCollapsed || !selection.rangeCount) {
+    if (!selection || selection.isCollapsed || !selection.rangeCount || !selection.focusNode) {
         return null;
     }
 
     // Firefox allows to select multiple ranges in the document by using Ctrl/Cmd+click
     // We only care about the last range
-    return selection.getRangeAt(selection.rangeCount - 1);
-}
-
-export function getSelection(): Selection | null {
-    const range = getRange();
-
-    if (!range) {
-        return null;
-    }
+    let range = selection.getRangeAt(selection.rangeCount - 1);
 
     const containerEl = findClosestElWithClass(range.endContainer as Element, 'textLayer');
     const startPage = getPageNumber(range.startContainer as Element);
     const endPage = getPageNumber(range.endContainer as Element);
 
-    if (!containerEl || !startPage || !endPage || startPage !== endPage) {
+    if (!containerEl || !startPage || !endPage) {
         return null;
     }
 
-    return { containerRect: containerEl.getBoundingClientRect(), location: endPage, range };
+    const canCreate = startPage === endPage;
+    let location = endPage;
+
+    if (!canCreate) {
+        range = document.createRange();
+        range.setEnd(selection.focusNode, selection.focusOffset);
```

Comment:
Is `setStart` not needed? From the screenshots it looks like the virtual element gets set to the focusNode span rather than a rect representing the rect on the initiating page? Is that the intended UX?

Reply:
`range.collapse` will collapse the range to the end, so `setStart` is not necessary.
Getting a rect representing the rect on the initiating page has to iterate the range which is too complicated for just showing an error. I've discussed the UX with product and they think it's ok to position the popup at the focus point.